### PR TITLE
Changes of incremental import billing

### PR DESCRIPTION
**Pull request by rajithaalurims**
## Description
Changes to add billing to incremental import.
ImportJobMetricsNotification needs to be published with ImportMode information so on Health PaaS side, we will know to bill the import requests or not. This PR has changes to add ImportMode in the notification that will be handled in Handler that is on HP side.

## Related issues
Addresses [issue [103222](https://microsofthealth.visualstudio.com/Health/_workitems/edit/103222/)].

## Testing
Unit tests added

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Import/ImportTests.cs [hunk 1]
```diff
@@ -145,6 +146,103 @@ public async Task GivenIncrementalLoad_MultipleInputVersions_ResourceNotExisting
             Assert.Equal(GetLastUpdated("2002"), result.Resource.Meta.LastUpdated);
         }
 
+        [Fact]
+        public async Task GivenIncrementalImportInvalidResource_ThenErrorLogsShouldBeOutput_FailedRCountShouldMatch()
```
Comment (LTA-Thinking) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Import/ImportTests.cs:150]: Nit: Is the R in the test title a typo or suppose to be the work Resource? Also, the test name should be in Given/When/Then format.

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorJobResult.cs [hunk 2]
```diff
@@ -54,5 +54,10 @@ public class ImportOrchestratorJobResult
         public int CreatedJobCount { get; set; } // TODO: remove in stage 3
 
         public long? TotalSizeInBytes { get; set; } // TODO: remove in stage 3
+
+        /// <summary>
+        /// Import mode.
+        /// </summary>
+        public ImportMode ImportMode { get; set; }
```
Comment (SergeyGaluzo) [inline src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorJobResult.cs:61]: Is import mode available in the job definition?

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorJobResult.cs [hunk 3]
```diff
@@ -54,5 +54,10 @@ public class ImportOrchestratorJobResult
         public int CreatedJobCount { get; set; } // TODO: remove in stage 3
 
         public long? TotalSizeInBytes { get; set; } // TODO: remove in stage 3
```
Comment (SergeyGaluzo) [inline src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorJobResult.cs:56]: If you don't mind, can you remove these?

**Comment by SergeyGaluzo** [comment]
>         private async Task SendImportMetricsNotification(JobStatus jobStatus, JobInfo jobInfo, ImportOrchestratorJobResult currentResult)

JobInfo contains Status and Result

---
Refers to: src/Microsoft.Health.Fhir.SqlServer/Features/Operations/Import/ImportOrchestratorJob.cs:268 in c55f771. [](commit_id = c55f771ecb603d8a5b9808a0c52da4fca7a0e0e0, deletion_comment = False)

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Import/ImportTests.cs (hunk 1 above)
Comment (rajithaalurims) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Import/ImportTests.cs]: @LTA-Thinking  Yes, type. Changed to GivenIncrementalImportInvalidResource__WhenImportData_ThenErrorLogsShouldBeOutputAndFailedCountShouldMatch.

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorJobResult.cs (hunk 2 above)
Comment (rajithaalurims) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorJobResult.cs]: @SergeyGaluzo  removed

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorJobResult.cs [hunk 4]
```diff
@@ -50,9 +50,5 @@ public class ImportOrchestratorJobResult
         public long SucceedImportCount { get; set; } // TODO: remove in stage 3
 
         public long FailedImportCount { get; set; } // TODO: remove in stage 3
-
-        public int CreatedJobCount { get; set; } // TODO: remove in stage 3
-
-        public long? TotalSizeInBytes { get; set; } // TODO: remove in stage 3
     }
```
Comment (SergeyGaluzo) [inline src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorJobResult.cs:53]: Any reason you did not remove these 2?

**Review by SergeyGaluzo — APPROVED** [review]
:shipit:

**Review by LTA-Thinking — DISMISSED** [review]

**Review by LTA-Thinking — APPROVED** [review]

**Review by fhibf — APPROVED** [review]

**Pull request merged**